airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/operation/AbstractStreamOperation.kt

```diff
@@ -27,8 +27,13 @@ abstract class AbstractStreamOperation<DestinationState : MinimumDestinationStat
     private val isTruncateSync: Boolean
     private val rawTableSuffix: String
     private val finalTmpTableSuffix: String
-    // null for truncate syncs, where we don't care at all about the initial status.
-    private val initialRawTableStatus: InitialRawTableStatus?
+    /**
+     * The status of the raw table that "matters" for this sync. Specifically:
+     * * For normal syncs / merge refreshes, this is the status of the real raw table)
+     * * For truncate refreshes, this is the status of the temp raw table (because we never even
+     * look at the real raw table)
+     */
+    private val initialRawTableStatus: InitialRawTableStatus
```

Comment:
@edgao  Remind me the difference here exactly. I really tried to follow this morning. :)

* normal sync/merge refresh: we populate a running raw table, then type-dedupe into the tmp table, then swap?
* for truncate refreshes: we populate a tmp raw table, then swap?

Reply:
for an overwrite, we populate a tmp raw table, truncate and dedupe into a tmp final table, and them swap both with the raw and final tables.
For an append, we don't do tmp tables.

truncate refresh is an overwrite. Everything else is an append